WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -88,18 +93,38 @@ - (void)fetchPostsForTopic:(ReaderAbstractTopic *)topic
                                   }];
 }
 
+
+- (void)updateTopic:(NSManagedObjectID *)topicID withAlgorithm:(NSString *)algorithm
+{
+    NSError *error;
+    ReaderAbstractTopic *topic = (ReaderAbstractTopic *)[self.managedObjectContext existingObjectWithID:topicID error:&error];
+    topic.algorithm = algorithm;
+
+    [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+}
+
+
 - (void)fetchPostsForTopic:(ReaderAbstractTopic *)topic
                earlierThan:(NSDate *)date
            deletingEarlier:(BOOL)deleteEarlier
                    success:(void (^)(NSInteger count, BOOL hasMore))success
                    failure:(void (^)(NSError *error))failure
 {
+    // Don't pass the algorithm if fetching a brand new list.
+    NSString *reqAlgorithm = [date isEqualToDate:[NSDate date]] ? nil : topic.algorithm;
```

Comment:
Something about `[date isEqualToDate:[NSDate date]]` here makes me uncomfortable... Maybe it doesn't read clearly why evaluating the passed `date` as equal to the current date would be the case in which we are fetching a brand new list. Maybe just some more documentation here?